src/stacked_borrows.rs

```diff
@@ -340,16 +339,10 @@ impl<'tcx> Stack {
                 tag,
             )))?;
 
-        // We must make sure there are no protected items remaining on the stack.
-        // Also clear the stack, no more accesses are possible.
-        for item in self.borrows.drain(..) {
-            if let Some(call) = item.protector {
-                if global.is_active(call) {
-                    return err!(MachineError(format!(
-                        "deallocating with active protector ({})", call
-                    )))
-                }
-            }
+        // Step 2: Remove all items.  Also checks for protectors.
+        while self.borrows.len() > 0 {
```

Comment:
```suggestion
        for item in self.borrows.drain(..).rev() {
```

Reply:
Same here, this is for symmetry with the code in `access`.